@mkouba can you fix the formatting issue:
```
Caused by: org.apache.maven.plugin.MojoFailureException: File '/home/runner/work/quarkus/quarkus/independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java' has not been previously formatted.  Please format file and commit before running validation!
```

--- a/independent-projects/qute/core/src/main/java/io/quarkus/qute/Parser.java
+++ b/independent-projects/qute/core/src/main/java/io/quarkus/qute/Parser.java
@@ -241,7 +241,7 @@ class Parser implements Function<String, Expression> {
 
     private void flushTag() {
         state = State.TEXT;
-        String content = buffer.toString();
+        String content = buffer.toString().trim();
         String tag = START_DELIMITER + content + END_DELIMITER;
 
         if (content.charAt(0) == Tag.SECTION.command) {

--- a/independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java
+++ b/independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java
@@ -163,6 +163,13 @@ public class ParserTest {
                 "Parser error on line 1: unterminated string literal or composite parameter detected for [#if (foo || bar]", 1);
     }
 
+    @Test
+    public void testWhitespace() {
+        Engine engine = Engine.builder().addDefaults().build();
+        assertEquals("Hello world", engine.parse("{#if true  }Hello {name }{/if  }").data("name", "world").render());
+        assertEquals("Hello world", engine.parse("Hello {name ?: 'world'  }").render());
+    }
+
     private void assertParserError(String template, String message, int line) {
         Engine engine = Engine.builder().addDefaultSectionHelpers().build();
         try {

